test/controllers/auth/SelfAssessmentStatusActionSpec.scala

```diff
@@ -95,32 +100,63 @@ class SelfAssessmentStatusActionSpec
         None,
         FakeRequest()
       )
+
       val result = harness()(request)
-      contentAsString(result) must include("NotYetActivatedOnlineFilerSelfAssessmentUser")
+      contentAsString(result) must include(s"NotYetActivatedOnlineFilerSelfAssessmentUser(${saUtr.utr})")
       verify(mockCitizenDetailsService, times(0)).getMatchingDetails(any())(any())
     }
   }
 
   "A user without an SA enrolment must" - {
     "when CitizenDetails has a matching SA account" - {
-      "return AmbiguousFilerSelfAssessmentUser" in {
-        implicit val request: AuthenticatedRequest[AnyContent] =
-          AuthenticatedRequest(
-            Some(Nino("AB123456C")),
-            None,
-            Credentials("", "Verify"),
-            ConfidenceLevel.L200,
-            None,
-            None,
-            None,
-            None,
-            FakeRequest())
+      "and they have an enrolment on another credential" - {
+        "return WrongCredentialsSelfAssessmentUser" in {
+          implicit val request: AuthenticatedRequest[AnyContent] =
+            AuthenticatedRequest(
+              Some(Nino("AB123456C")),
+              None,
+              Credentials("", "Verify"),
+              ConfidenceLevel.L200,
+              None,
+              None,
+              None,
+              None,
+              FakeRequest())
+
+          val saUtr = SaUtr("1111111111")
+
+          when(mockCitizenDetailsService.getMatchingDetails(any())(any()))
+            .thenReturn(Future.successful(MatchingDetailsSuccessResponse(MatchingDetails(Some(saUtr)))))
+          when(enrolmentsConnector.getUserIdsWithEnrolments(meq(saUtr.utr))(any(), any()))
+            .thenReturn(Future.successful(Seq("some cred id")))
+
+          val result = harness()(request)
+          contentAsString(result) must include(s"WrongCredentialsSelfAssessmentUser(${saUtr.utr})")
+        }
+      }
 
-        when(mockCitizenDetailsService.getMatchingDetails(any())(any()))
-          .thenReturn(Future.successful(MatchingDetailsSuccessResponse(MatchingDetails(Some(SaUtr("1111111111"))))))
-        val result = harness()(request)
-        contentAsString(result) must include("AmbiguousFilerSelfAssessmentUser")
-        verify(mockCitizenDetailsService, times(1)).getMatchingDetails(any())(any())
+      "and they have no SA enrolment" - {
+        "return NotEnrolledSelfAssessmentUser" in {
+          implicit val request: AuthenticatedRequest[AnyContent] =
+            AuthenticatedRequest(
```

Comment:
You've got a few of these requests where they're pretty much identical bar 1/2 fields. Can you just refactor those away so we only see the relevant changes to the request?